R/Sound_DMR.R

```diff
@@ -974,74 +974,89 @@ changepoint_analysis <- function(whole_df,
                                  CHH_penalty = int,
                                  target_genes = c(),
                                  save_plots = FALSE,
-                                 z_col = 'column') {
+                                 z_col = 'column',
+                                 subset_size = 35000) {
+  # Create the list of subset groups
+  if (length(unique(whole_df$Gene)) > 1) {
+    subset_col <- "Gene"
+    whole_df['row_num'] = NA
+    whole_df['group_num'] = NA
+  } else if (length(unique(whole_df$Gene)) == 1) {
+    whole_df <- whole_df %>%
+      mutate(row_num = row_number(),
+             group_num = (row_num %/% subset_size) + 1)
+    subset_col <- "group_num"
+    target_genes <- unique(whole_df$group_num)
+  }
+  
+  subsets <- unique(whole_df[[subset_col]])
+  large_genes <- c()
+  
+  # Create the structure of the final df
   everything <- tibble::as_tibble(matrix(ncol = 8))
```

Comment:
nit : I think we can change the name the variable 'everything' here.

Reply:
For sure!